presto-oracle/src/main/java/io/prestosql/plugin/oracle/OracleClientModule.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.oracle;
+
+import com.google.inject.Binder;
+import com.google.inject.Provides;
+import com.google.inject.Scopes;
+import com.google.inject.Singleton;
+import io.airlift.configuration.AbstractConfigurationAwareModule;
+import io.prestosql.plugin.jdbc.BaseJdbcConfig;
+import io.prestosql.plugin.jdbc.ConnectionFactory;
+import io.prestosql.plugin.jdbc.DriverConnectionFactory;
+import io.prestosql.plugin.jdbc.JdbcClient;
+import io.prestosql.plugin.jdbc.credential.CredentialProvider;
+import oracle.jdbc.OracleDriver;
+
+import java.sql.SQLException;
+import java.util.Properties;
+
+import static io.airlift.configuration.ConfigBinder.configBinder;
+
+public class OracleClientModule
+        extends AbstractConfigurationAwareModule
+{
+    @Override
+    protected void setup(Binder binder)
+    {
+        binder.bind(JdbcClient.class).to(OracleClient.class).in(Scopes.SINGLETON);
+        configBinder(binder).bindConfig(BaseJdbcConfig.class);
+        configBinder(binder).bindConfig(OracleConfig.class);
+    }
+
+    @Provides
+    @Singleton
+    public static ConnectionFactory connectionFactory(BaseJdbcConfig config, CredentialProvider credentialProvider, OracleConfig oracleConfig)
+            throws SQLException
+    {
+        Properties connectionProperties = new Properties();
+        connectionProperties.setProperty("defaultRowPrefetch", String.valueOf(oracleConfig.getDefaultRowPrefetch()));
+        connectionProperties.setProperty("defaultBatchValue", String.valueOf(oracleConfig.getDefaultBatchValue()));
+        connectionProperties.setProperty("processEscapes", String.valueOf(oracleConfig.isProcessEscapes()));
```

Comment:
why do we need this?

Reply:
the configurations are based on this https://docs.oracle.com/cd/E11882_01/appdev.112/e13995/oracle/jdbc/OracleDriver.html

are implemented to allow the configuration, if not necessary they can be removed without problem

Reply:
I'm not sure of the other options... 
Synonyms should be provided as an option. The JDBC driver automatically resolves *some* synonyms.
Most organizations are used to accessing their database tables through synonyms - supporting this is important to transparent ease of use. 
If you don't want the performance impact or have no intention of using it, you can disable it.

Reply:
@bendemott's comment is for `includeSynonyms` on the next line. I'm responding to `processEscapes`.

This is a standard JDBC feature [explained here](https://docs.oracle.com/en/database/oracle/oracle-database/19/jjdbc/JDBC-reference-information.html#GUID-FCB7E652-4532-47AF-9783-B7E2B6ADA41C). We should remove this config, as this is not something end users should configure, as they do not control the SQL generated by the connector.

Rather than adding a driver-specific property, we should instead call `setEscapeProcessing(false)` on `Statement` (and `PreparedStatement`) instances we create in the `presto-base-jdbc` library, since we don't generate SQL with escapes. This would be a separate change from this PR.

Reply:
config removed